### PR TITLE
Add cell metrics to resultstore

### PR DESCRIPTION
**Pull request by bryan-lou**
Finishing up Sultan's PR: #1253

### CODE REVIEW

File: pkg/updater/resultstore/resultstore.go [hunk 1]
```diff
@@ -536,6 +538,59 @@ func processGroup(tg *configpb.TestGroup, group *invocationGroup) *updater.Infla
...
+			}
+		} else {
+			testResultProperties(properties, test.GetTestSuite())
+		}
+	}
+}
+
+// testResultDuration calculates the overall duration of a target, converts
+// milliseconds to minutes and returns the duration.
+func testResultDuration(target *resultstorepb.Target) int64 {
```
Comment (michelle192837) [inline pkg/updater/resultstore/resultstore.go:581]: Comment from https://github.com/GoogleCloudPlatform/testgrid/pull/1253 still applies; I think this should return a time.Duration, or should specify milliseconds (not just in the comment, but in the function name or similar).

It also looks like this no longer returns the duration for the target's test results, just the target?

**Review by michelle192837 — COMMENTED** [review]
Overall: Please note the PR this is based on when carrying over someone else's change, and credit the original author.

In this case, I might include 2 commits: the one you're basing it off of, and a commit with your changes on top of that, so it's easier to review.

File: pkg/updater/resultstore/resultstore.go (hunk 1 above)
Comment (bryan-lou) [reply pkg/updater/resultstore/resultstore.go]: Fixed!

File: pkg/updater/gcs.go [hunk 2]
```diff
@@ -529,6 +529,9 @@ func overallCell(result gcsResult) Cell {
 // ElapsedKey is the key for the test duration metric.
 const ElapsedKey = "test-duration-minutes"
 
+// TestMethodsElapsedKey is an automatically-added metric for sponge.
```
Comment (michelle192837) [inline pkg/updater/gcs.go:532]: nit: Would match this to "ElapsedKey" above for consistency.

File: pkg/updater/resultstore/resultstore.go [hunk 3]
```diff
@@ -538,18 +538,22 @@ func processGroup(tg *configpb.TestGroup, group *invocationGroup) *updater.Infla
...
 	testResultProperties(properties, sar.ActionProto.GetTestAction().GetTestSuite())
 	numerics := updater.Means(properties)
+	targetElapsed := sar.TargetProto.GetTiming().GetDuration().AsDuration()
+	if targetElapsed > 0 {
+		numerics[updater.ElapsedKey] = float64(targetElapsed.Milliseconds()) / 60000
+	}
 
 	if dur := testResultDuration(sar.ActionProto.GetTestAction().GetTestSuite()); dur > 0 {
-		numerics[updater.ElapsedKey] = float64(dur) / 60000
+		numerics[updater.TestMethodsElapsedKey] = float64(dur.Milliseconds()) / 60000
```
Comment (michelle192837) [inline pkg/updater/resultstore/resultstore.go:553]: nit: I think you don't need to do milliseconds conversion, you can just take Minutes() instead?

File: pkg/updater/resultstore/resultstore_test.go
Comment (michelle192837) [inline pkg/updater/resultstore/resultstore_test.go:1]: I think the unit tests need an update; it looks like some of them are meant to test picking up durations, but don't actually test that now.

File: pkg/updater/gcs.go (hunk 2 above)
Comment (bryan-lou) [reply pkg/updater/gcs.go]: done

File: pkg/updater/resultstore/resultstore_test.go
Comment (bryan-lou) [reply pkg/updater/resultstore/resultstore_test.go:1]: updated!

File: pkg/updater/gcs.go [hunk 4]
```diff
@@ -526,9 +526,12 @@ func overallCell(result gcsResult) Cell {
 	return c
 }
 
-// ElapsedKey is the key for the test duration metric.
+// ElapsedKey is the key for the test results duration metric.
```
Comment (michelle192837) [inline pkg/updater/gcs.go:529]: nit: test-duration-minutes should be the test / target duration.

The one below, for consistency, should be "test-methods-duration-minutes", for the test methods / test results.

Comment (michelle192837) [reply pkg/updater/gcs.go]: Sorry to nitpick, I just think consistency with the codebase will be helpful in the long run.

Can this get updated to:

```
// ElapsedKey is the key for the target duration metric.
const ElapsedKey = "test-duration-minutes"

// TestMethodsElapsedKey is the key for the test results duration metric.
const TestMethodsElapsedKey = "test-methods-duration-minutes"
```

Thanks! Looks good to go after that.

Comment (bryan-lou) [reply pkg/updater/gcs.go]: Fixed! @michelle192837

**Review by michelle192837 — APPROVED** [review]

**Pull request merged**